sys/include/atomic_utils.h

```diff
@@ -753,9 +774,31 @@ static inline atomic_bit_u32_t atomic_bit_u32(volatile uint32_t *dest,
  * @brief   Create a reference to a bit in an `uint64_t`
  * @param[in]       dest        Memory containing the bit
  * @param[in]       bit         Bit number (`0` refers to the least significant)
+ *
+ * @return Opaque reference to the bit.
  */
 static inline atomic_bit_u64_t atomic_bit_u64(volatile uint64_t *dest,
                                               uint8_t bit);
+
+/**
+ * @brief   Create a reference to a bit in an `unsigned int`
+ * @param[in]       dest        Memory containing the bit
+ * @param[in]       bit         Bit number (`0` refers to the least significant)
+ *
+ * @return Opaque reference to the bit.
+ */
+static inline atomic_bit_unsigned_t atomic_bit_unsigned(volatile unsigned *dest,
+                                                        uint8_t bit)
+{
+#if UINT_MAX == UINT16_MAX
+    return atomic_bit_u16(dest, bit);
+#elif UINT_MAX == UINT32_MAX
+    /* Some archs define uint32_t as unsigned long, we need to cast. */
+    return atomic_bit_u32((uint32_t volatile *)dest, bit);
+#else
+    return atomic_bit_u64(dest, bit);
```

Comment:
```suggestion
    return atomic_bit_u64((volatile uint64_t)dest, bit);
```

I don't have a practical example, but I do believe that `unsigned int`, `unsigned long`, and `unsigned long long` could in full compliance with the C standard all be 64 bit.